src/Google.Cloud.Metadata.V1/MetadataClient.cs

```diff
@@ -0,0 +1,571 @@
+        public virtual TokenResponse GetAccessToken()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Gets a token response which contains an access token to authorize a request asynchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <exception cref="TokenResponseException">A status code other than success was returned from the server.</exception>
+        /// <returns>The token response.</returns>
+        public virtual Task<TokenResponse> GetAccessTokenAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Gets the instance's custom metadata value with the specified key synchronously.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Custom metadata are just opaque key/value pairs which allow for custom configuration parameters, scripts, or other
+        /// relatively small pieces of data to be associated with the instance.
+        /// </para>
+        /// <para>
+        /// The key must conform to the following regular expression: [a-zA-Z0-9-_]+ and must be less than 128 bytes in length.
+        /// </para>
+        /// </remarks>
+        /// <param name="key">The key of the instance's custom metadata value to get.</param>
+        /// <exception cref="ArgumentException">The key is not in the proper format.</exception>
+        /// <seealso cref="InstanceMetadata.CustomMetadata"/>
+        public virtual void GetCustomInstanceMetadata(string key)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Gets the instance's custom metadata value with the specified key asynchronously.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Custom metadata are just opaque key/value pairs which allow for custom configuration parameters, scripts, or other
+        /// relatively small pieces of data to be associated with the instance.
+        /// </para>
+        /// <para>
+        /// The key must conform to the following regular expression: [a-zA-Z0-9-_]+ and must be less than 128 bytes in length.
+        /// </para>
+        /// </remarks>
+        /// <param name="key">The key of the instance's custom metadata value to get.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <exception cref="ArgumentException">The key is not in the proper format.</exception>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        /// <seealso cref="InstanceMetadata.CustomMetadata"/>
+        public virtual Task GetCustomInstanceMetadataAsync(string key, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
No, this should have been `Task\<string\>`. It is just meant to get a single value asynchronously.